### ISSUE TITLE
Unable to Create Hook for User Service

### ISSUE THREAD

**Issue by lnguyenfx** [feathersjs/feathers#295]
I'm trying to create the gravatar hook for the user service by following [this step](http://docs.feathersjs.com/getting-started/formatting-and-hooks.html#gravatar-profile-images) from section 3.3 of the getting-started tutorial, but after specifying the method the hook is for, no files were created.

FYI I'm using Node on Windows

**Comment by daffl** [comment]
Are you able to create other hooks? Did it throw an error?

**Comment by lnguyenfx** [comment]
Yes, I am able to create hooks for other services, but not for the User service. There was no error when I used the "feathers generate hook" command, but when I used the "yo feathers:hook" command, I received the following error:

`Error: Could not find an array for object key create to insert testHook()
    at Object.exports.addToArrayInObject (C:\Users\LN\AppData\Roaming\npm\node_modules\generator-feathers\lib\transform.
js:72:11)
    at C:\Users\LN\AppData\Roaming\npm\node_modules\generator-feathers\generators\hook\index.js:19:17
    at Array.forEach (native)
    at importHook (C:\Users\LN\AppData\Roaming\npm\node_modules\generator-feathers\generators\hook\index.js:18:13)
    at module.exports.generators.Base.extend.writing (C:\Users\LN\AppData\Roaming\npm\node_modules\generator-feathers\ge
nerators\hook\index.js:129:5)
    at Object.<anonymous> (C:\Users\LN\AppData\Roaming\npm\node_modules\generator-feathers\node_modules\yeoman-generator
\lib\base.js:436:25)
    at C:\Users\LN\AppData\Roaming\npm\node_modules\generator-feathers\node_modules\run-async\index.js:26:25
    at C:\Users\LN\AppData\Roaming\npm\node_modules\generator-feathers\node_modules\run-async\index.js:25:19
    at C:\Users\LN\AppData\Roaming\npm\node_modules\generator-feathers\node_modules\yeoman-generator\lib\base.js:448:8
    at tryOnImmediate (timers.js:534:15)`

My generator-feathers version is 0.8.3

**Comment by daffl** [comment]
That does sound like a bug in the generator AST transformations. How does your user hook file look like?

**Comment by lnguyenfx** [comment]
```javascript
'use strict';

const gravatar = require('./gravatar');

const globalHooks = require('../../../hooks');
const hooks = require('feathers-hooks');
const auth = require('feathers-authentication').hooks;

exports.before = {
  all: [],
  find: [
    auth.verifyToken(),
    auth.populateUser(),
    auth.restrictToAuthenticated()
  ],
  get: [
    auth.verifyToken(),
    auth.populateUser(),
    auth.restrictToAuthenticated(),
    auth.restrictToOwner({ ownerField: 'id' })
  ],
  create: [
    auth.hashPassword(),
    gravatar()
  ],
  update: [
    auth.verifyToken(),
    auth.populateUser(),
    auth.restrictToAuthenticated(),
    auth.restrictToOwner({ ownerField: 'id' })
  ],
  patch: [
    auth.verifyToken(),
    auth.populateUser(),
    auth.restrictToAuthenticated(),
    auth.restrictToOwner({ ownerField: 'id' })
  ],
  remove: [
    auth.verifyToken(),
    auth.populateUser(),
    auth.restrictToAuthenticated(),
    auth.restrictToOwner({ ownerField: 'id' })
  ]
};

exports.after = {
  all: [hooks.remove('password')],
  find: [],
  get: [],
  create: [],
  update: [],
  patch: [],
  remove: []
};
```

**Comment by daffl** [comment]
Thank you! I found the issue and it is fixed in the generator v0.8.4
